src/test/java/org/assertj/core/internal/maps/Maps_assertContainsOnlyKeys_iterable_Test.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.internal.maps;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.data.MapEntry.entry;
+import static org.assertj.core.error.ShouldContainOnlyKeys.shouldContainOnlyKeys;
+import static org.assertj.core.internal.ErrorMessages.*;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.MapsBaseTest;
+import org.assertj.core.test.Maps;
+import org.assertj.core.util.Streams;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for
+ * <code>{@link org.assertj.core.internal.Maps#assertContainsOnlyKeys(AssertionInfo, Map, Iterable<Object>)}</code>
+ * .
+ *
+ * @author Sebastian Kempken
+ */
+public class Maps_assertContainsOnlyKeys_iterable_Test extends MapsBaseTest {
+
+  private final String PLACEHOLDER_FOR_ERROR_MESSAGE = "keys iterable";
+
+  @Test
+  public void should_fail_if_actual_is_null() {
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> maps.assertContainsOnlyKeys(someInfo(), null,
+                                                                                                 Arrays.asList("name")))
+                                                   .withMessage(actualIsNull());
+  }
+
+  @Test
+  public void should_fail_if_given_keys_array_is_null() {
+    assertThatNullPointerException().isThrownBy(() -> maps.assertContainsOnlyKeys(someInfo(), actual,
+                                                                                  (Iterable<String>) null))
+                                    .withMessage(keysToLookForIsNull(PLACEHOLDER_FOR_ERROR_MESSAGE));
+  }
+
+  @Test
+  public void should_fail_if_given_keys_array_is_empty() {
+    assertThatIllegalArgumentException().isThrownBy(() -> maps.assertContainsOnlyKeys(someInfo(), actual,
+                                                                                      Arrays.asList(emptyKeys())))
```

Comment:
`Arrays.asList(emptyKeys())` by `list()` or `Collections.emptyList()` (and static import it)

Reply:
done